### PR TITLE
Python2.x and python3.x compatibility patches on develop branch

### PR DESCRIPTION
**Pull request by samvarankashyap**
$subject

### CODE REVIEW

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
@14rcole Any idea what happened to downstream tests?

**Comment by samvarankashyap** [comment]
[test]

**Comment by 14rcole** [comment]
@samvarankashyap no, it seems to be a networking issue but I couldn't really say more than that

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
Downstream failure due to outdated openstack Pinfiles
rebase should fix the problem.

**Comment by samvarankashyap** [comment]
[test]

File: config/Dockerfiles/tests.d/openstack/04_os-sg-new [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash -xe
 
 # Verify the openstack security group provisioning with template data
-# distros.exclude: none
+# distros.exclude: centos7
```
Comment (14rcole) [inline config/Dockerfiles/tests.d/openstack/04_os-sg-new]: why are we excluding centos?

Comment (samvarankashyap) [reply config/Dockerfiles/tests.d/openstack/04_os-sg-new]: @14rcole The account we are currently using to provision security groups is hitting its quota limit when we provision 3 security groups in parallel. In order to minimize the number of security groups I excluded centos7

**Comment by samvarankashyap** [comment]
Rebased to resolve conflicts with recent merges

**Comment by samvarankashyap** [comment]
[test] failing on libvirt-network provisioning retesting

**Comment by samvarankashyap** [comment]
[test]

**Comment by samvarankashyap** [comment]
[merge]

**Pull request merged**